### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by MishaLehotskyi**
[DEMO LINK](https://MishaLehotskyi.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -32,172 +68,128 @@ export const App: React.FC = () => {
...
-          {/* don't show this button if there are no completed todos */}
-          <button
-            type="button"
-            className="todoapp__clear-completed"
-            data-cy="ClearCompletedButton"
-          >
-            Clear completed
-          </button>
-        </footer>
+        {todos[0] && (
```
Comment (etojeDenys) [inline src/App.tsx:71]: you can just check if the todos array is not empty using the `.length`

File: src/App.tsx [hunk 2]
```diff
@@ -32,172 +68,128 @@ export const App: React.FC = () => {
...
+          <>
+            <section className="todoapp__main" data-cy="TodoList">
+              {filteredTodos.map(value => (
+                <div
+                  data-cy="Todo"
```
Comment (etojeDenys) [inline src/App.tsx:76]: move this into TodoList component

File: src/App.tsx [hunk 3]
```diff
@@ -32,172 +68,128 @@ export const App: React.FC = () => {
...
+
+                  {/* overlay will cover the todo while it is being updated */}
+                  <div data-cy="TodoLoader" className="modal overlay">
+                    <div
+                      className="modal-background has-background-white-ter"
+                    />
+                    <div className="loader" />
+                  </div>
+                </div>
+              ))}
```
Comment (etojeDenys) [inline src/App.tsx:112]: all this you can move into TodoItem component

File: src/App.tsx [hunk 4]
```diff
@@ -1,10 +1,46 @@
...
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 11983;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState('all');
```
Comment (etojeDenys) [inline src/App.tsx:13]: better to create some enum for status values

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 5]
```diff
@@ -1,10 +1,48 @@
...
+
+  const filteredTodos = todos.filter(todo => {
+    switch (status) {
+      case Status.ACTIVE:
+        return !todo.completed;
+      case Status.COMPLETED:
+        return todo.completed;
+      case Status.ALL:
+      default:
+        return true;
```
Comment (loralevitska) [inline src/App.tsx:40]:
```suggestion
      case Status.ACTIVE:
        return !todo.completed;
        
      case Status.COMPLETED:
        return todo.completed;
        
      case Status.ALL:
      default:
        return true;
```

File: src/App.tsx [hunk 6]
```diff
@@ -32,172 +70,88 @@ export const App: React.FC = () => {
...
-            data-cy="ClearCompletedButton"
-          >
-            Clear completed
-          </button>
-        </footer>
+        {!!todos.length && (
+          <>
+            <TodoList todos={filteredTodos} />
+
+            {/* Hide the footer if there are no todos */}
```
Comment (loralevitska) [inline src/App.tsx:77]: Remove comment. Try to write obvious code with meaningful variable and function naming
Fix all cases

File: src/App.tsx [hunk 7]
```diff
@@ -32,172 +70,88 @@ export const App: React.FC = () => {
...
+            <footer className="todoapp__footer" data-cy="Footer">
+              <span className="todo-count" data-cy="TodosCounter">
+                {`${todosCounter} items left`}
+              </span>
+
+              {/* Active filter should have a 'selected' class */}
+              <nav className="filter" data-cy="Filter">
```
Comment (loralevitska) [inline src/App.tsx:84]: It could be common component

File: src/components/TodoList/TodoList.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+interface Props {
+  todos: Todo[],
+}
+
+export const TodoList: React.FC<Props> = ({ todos }) => (
+  <section className="todoapp__main" data-cy="TodoList">
```
Comment (loralevitska) [inline src/components/TodoList/TodoList.tsx:10]: Use semantic tags for list `ul` `li`. Don't forget to remove default `li` dots and `ul` paddings

**Review by loralevitska — APPROVED** [review]
Well done)

**Pull request closed without merging**